components/camel-zipkin/src/main/java/org/apache/camel/zipkin/ZipkinServerResponseAdapter.java

```diff
@@ -45,7 +45,7 @@ void onResponse(Exchange exchange, SpanCustomizer span) {
         span.tag("camel.server.exchange.id", id);
         span.tag("camel.server.exchange.pattern", mep);
 
-        if (exchange.getException() != null) {
+        if (exchange.getException() != null && exchange.getException().getMessage() != null) {
```

Comment:
There is still a failure, so you should instead if the message is null, create a empty message, or use the exception classname as the message.

Reply:
So I'll roll back that if statement and if _message_ is null (or blank???) I will change it to exception classname, is that right?

Reply:
Yes correct, then we always ensure that if an exception occurred then zipkin gets that as a failure event

Reply:
Change request done.